src/util/BaseLoginRouter.js

```diff
@@ -31,7 +32,7 @@ define([
   'util/Bundles',
   'util/Logger'
 ],
-function (Okta, BrowserFeatures, Settings,
+function (Q, Okta, BrowserFeatures, Settings,
```

Comment:
Curious why this change is preferred over what we had before.

Reply:
I don't remember exactly. My guess is it has to do with unhandled promise rejection? I'll change it back and see what effect it has on the tests.

Reply:
There is a mixed usage of Q and standard promise. I am adding another commit which changes `fail` to `catch` in these cases so we can accept either.